pkg/operator/boundsatokensignercontroller/controller.go

```diff
@@ -0,0 +1,402 @@
+		}
+	}
+	return nil
+}
+
+// ensurePublicKeyConfigMap ensures that the public key in the operator secret is
+// present in the operand configmap. If the configmap is missing, it will be created
+// with the current public key. If the configmap exists but does not contain the
+// current public key, the key will be added.
+func (c *BoundSATokenSignerController) ensurePublicKeyConfigMap() error {
+	// Retrieve the operator secret that contains the current public key
+	operatorSecret, err := c.secretClient.Secrets(operatorNamespace).Get(NextSigningKeySecretName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the configmap that needs to contain the current public key
+	configMap, err := c.configMapClient.ConfigMaps(targetNamespace).Get(PublicKeyConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// Initialize a new configmap if it was missing
+	if errors.IsNotFound(err) {
+		configMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: targetNamespace,
+				Name:      PublicKeyConfigMapName,
+			},
+			Data: map[string]string{},
+		}
+	}
+
+	currPublicKey := string(operatorSecret.Data[PublicKeyKey])
+	hasKey := configMapHasValue(configMap, currPublicKey)
+	if !hasKey {
+		// Increment until a unique name is found to ensure that the new public key
+		// does not overwrite an existing one. Except where key revocation is
+		// involved (which would require manual deletion of the verifying public
+		// key), existing public keys in the configmap should be maintained to
+		// minimize the potential for not being able to validate issued tokens.
+		nextKeyIndex := len(configMap.Data) + 1
+		nextKeyKey := ""
+		for len(nextKeyKey) == 0 {
+			possibleKey := fmt.Sprintf("service-account-%03d.pub", nextKeyIndex)
+			_, ok := configMap.Data[possibleKey]
+			if !ok {
+				nextKeyKey = possibleKey
+				break
+			}
+			nextKeyIndex += 1
+		}
+
+		// Ensure the configmap is updated with the current public key
+		configMap.Data[nextKeyKey] = currPublicKey
+		configMap, _, err = resourceapply.ApplyConfigMap(c.configMapClient, c.eventRecorder, configMap)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureOperandSigningSecret ensures that the current signing key is
+func (c *BoundSATokenSignerController) ensureOperandSigningSecret() error {
+	// Retrieve the operator signing secret
+	operatorSecret, err := c.secretClient.Secrets(operatorNamespace).Get(NextSigningKeySecretName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the operand signing secret
+	operandSecret, err := c.secretClient.Secrets(targetNamespace).Get(SigningKeySecretName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	// If operand secret matches the operator secret, all done
+	operandSecretUpToDate := (operandSecret != nil &&
+		bytes.Equal(operandSecret.Data[PublicKeyKey], operatorSecret.Data[PublicKeyKey]) &&
```

Comment:
The operator secret can never be nil since all errors on retrieval (including `NotFound`) result  in exit. The operand secret is checked for nil. Am I missing something?

Reply:
ah, my bad, I managed to overlook the statement on the above line. Sorry